src/DiscoveryClient.coffee

```diff
@@ -48,29 +71,68 @@ class DiscoveryClient
         else
           Promise.reject(e)
 
+  connect:(callback) =>
+    @_connect(callback, @discoveryConnector, @serverList, @regionsConfig.discoverRegion)
+
+    for region, connectors of @regions
+      @_connect(null, connectors.discoveryConnector, connectors.serverList, region)
+
+  getUrlFromRegion:(baseUrl, regionName ) ->
+    baseUrl.replace '*', regionName
+ 
+  getHostUrl: () =>
+    @getUrlFromRegion @host, @regionsConfig.discoverRegion
+
   reconnect:(callback)=>
     @connect(callback)
 
   stopAnnouncementHeartbeat:()=>
-    if @heartbeatIntervalId
-      clearInterval(@heartbeatIntervalId)
+    if @heartbeatIntervalIds
+      for intervals in @heartbeatIntervalIds
+        clearInterval(intervals)
+      @heartbeatIntervalIds = []
 
   startAnnouncementHeartbeat:()=>
     @stopAnnouncementHeartbeat()
+    if not @heartbeatIntervalIds
+      @heartbeatIntervalIds = []
 
-    @heartbeatIntervalId = setInterval(
+    @heartbeatIntervalIds.push setInterval(
       @discoveryAnnouncer.pingAllAnnouncements
       @HEARTBEAT_INTERVAL_MS
     )
+
+    for region,connectors of @regions
+      @heartbeatIntervalIds.push setInterval(
+        connectors.discoveryAnnouncer.pingAllAnnouncements
+        @HEARTBEAT_INTERVAL_MS
```

Comment:
maybe you could make this a function of `discoveryAnnouncer` (start/stop heartbeat)